integrations/spark/spark-3.1/openhouse-spark-runtime/src/main/scala/com/linkedin/openhouse/spark/sql/catalyst/constants/Principal.scala

```diff
@@ -0,0 +1,18 @@
+package com.linkedin.openhouse.spark.sql.catalyst.constants
+
+/**
+ * This object is used to represent keyword global user group "PUBLIC" which maps to the acl policy representation "*"
+ */
+object Principal {
+  private val GLOBAL_USER_GROUP = "PUBLIC"
+  private val GLOBAL_USER_GROUP_ACL = "*"
+  def apply(principal: String): String = principal match {
+    case GLOBAL_USER_GROUP => GLOBAL_USER_GROUP_ACL
+    case _ => principal
+  }
+
+  def unapply(principalAcl: String): Option[String] = principalAcl match {
```

Comment:
Do we ever need to use this?

Reply:
+1. If not used can be removed.

Reply:
This is a bit of Scala syntactic sugar, for this line:
```
 val Principal(principal) = aclPolicy.getPrincipal
 ```
 
 Which would map `principal` from `*` to `PUBLIC` under the hood, I thought it would look better like this since I used the reverse function already to do the mapping, I can make it more explicit if we would prefer that.

Reply:
lets make this more explicit, thanks